pkg/controller/namespace.go

```diff
@@ -0,0 +1,271 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+	"k8s.io/client-go/util/workqueue"
+
+	"github.com/cert-manager/istio-csr/cmd/app/options"
+)
+
+const (
+	resyncPeriod        = time.Second * 60
```

Comment:
recent cert-manager discussions linking to controller-runtime work indicates that this can be waaay longer ;)

Reply:
How come, are their concerns about syncing too much?

Could you expand a bit more?

Reply:
long story... becomes problematic when talking 10K resources